lib/chef/chef_fs/file_system/repository/chef_repository_file_system_root_dir.rb

```diff
@@ -109,7 +110,14 @@ def create_child(name, file_contents = nil)
             else
               child_paths[name].each do |path|
                 begin
-                  Dir.mkdir(path)
+                  events = Chef::EventDispatch::Dispatcher.new
+                  node = Chef::Node.new
+                  run_context = Chef::RunContext.new(node, {}, events)
+                  dir = Chef::Resource::Directory.new(path, run_context)
```

Comment:
indeed, it would be nice to have that kind of API, but we don't.  its easier for you to commit these kinds of atrocities because chef doesn't have direct deps on your code, so all the deps run one direction -- but here it runs both ways and that will only lead to loops and great sadness..